src/wallet/wallet2.cpp

```diff
@@ -5772,7 +5817,24 @@ void wallet2::store_to(const std::string &path, const epee::wipeable_string &pas
     // store should only exist if the MMS is really active
     m_message_store.write_to_file(get_multisig_wallet_state(), m_mms_file);
   }
-  
+}
+//----------------------------------------------------------------------------------------------------
+bool wallet2::get_cache_file_data(const epee::wipeable_string &password, wallet2::cache_file_data &cache_file_data)
+{
+  trim_hashchain();
+
+  std::stringstream oss;
+  boost::archive::portable_binary_oarchive ar(oss);
+  ar << *this;
```

Comment:
That's throwy for a function that returns a bool.

Reply:
I made get_cache_file_data() return boost::optional<wallet2::cache_file_data> to be consistent with get_keys_file_data(), and an exception is thrown if boost::none is returned.